### DISCUSSION TITLE
Help with initial setup?

### DISCUSSION THREAD

**Discussion by filmo003** [fevieira27/MoveToBeActive#4] [Q&A] [answered]
I really enjoy this watchface and wanted to see if I could maybe add some functionality/options or fork it for myself. I'm new to Garmin programming and am having difficulties getting the simulator running from the cloned repo. Specifically I'm having the following problems. 

There initially is no manifest.xml file.
I get an error for a missing Checkers.fnt file in /fonts.
Creating my own manifest.xml and fake Checkers.fnt file it compiles until it hits the following error `ERROR: venu2: unknown:0: Unable to resolve entry class. Check the manifest file.`

Any ideas? Thanks!

**Comment by fevieira27** [comment, marked as answer]
Hi! Have you seen this explanation from garmin about the manifest file? https://developer.garmin.com/connect-iq/core-topics/manifest-and-permissions/

The reason why I can't share it openly is that the manifest file has a unique Id that identifies it on the connectIQ store, but it is a very straightforward file to create that only includes: the unique ID, list of supported watches and required permissions. 

My guess is that you might be missing the product (venu 2 on your test) on the manifest. The easiest way to add the products is by using visual studio code's shortcuts (ctrl+shift+P, if I'm not wrong) and selecting add/edit products, then select the watches that you want to support and they will be automatically added to the manifest. Generate a new UID and changing permissions all could be changed the same way (through the shortcuts).

If you've tried this and still can't seem to run the code, then just post your manifest here (without the UID) and I could have a look at it.

**Reply by filmo003** [reply]
Here's the manifest I'm using. I'm able to do the different VS Code shortcuts for Monkey C (verify installation, edit application, set products, etc) so it's at least finding the manifest and adding/removing things to it. The "unable to resolve entry class" error leads me to believe it could maybe be the `entry=MoveToBeActiveApp` line?


```
<?xml version="1.0"?>
<!-- This is a generated file. It is highly recommended that you DO NOT edit this file. -->
<iq:manifest version="3" xmlns:iq="http://www.garmin.com/xml/connectiq">
    <!--
        Use "Monkey C: Edit Application" from the Visual Studio Code command palette
        to update the application attributes.
    -->
    <iq:application id="<unique-UUID>" type="watchface" name="@Strings.AppName" entry="MoveToBeActiveApp" launcherIcon="@Drawables.LauncherIcon" minApiLevel="4.0.0">
        <!--
            Use the following from the Visual Studio Code comand palette to edit
            the build targets:
            "Monkey C: Set Products by Version" - Lets you add all producs that meet the
                                                    minimum SDK version of the application
            "Monkey C: Edit Products" - Lets you add or remove any product
        -->
        <iq:products>
            <iq:product id="fenix7"/>
            <iq:product id="venu2"/>
        </iq:products>
        <!--
            Use "Monkey C: Edit Permissions" from the Visual Studio Code command
            palette to update permissions.
        -->
        <iq:permissions/>
        <!--
            Use "Monkey C: Edit Languages" from the Visual Studio Code command
            palette to edit your compatible language list.
        -->
        <iq:languages/>
        <!--
            Use "Monkey C: Configure Monkey Barrel" from the Visual Studio Code
            command palette to edit the included barrels.
        -->
        <iq:barrels/>
    </iq:application>
</iq:manifest>
```

**Reply by fevieira27** [reply]
A few things that you could try:
- Add the permissions: position,  user profile and sensor history
- Add at least one language
- I believe that the application entry needs to have the same name as the class from the main app, which in this case would be AnalogWatch

Unfortunately, I'm travelling this week and without access to visual studio to test your manifest or compare it with mine, but those are the things that I could remember right now. Hope it helps!

**Reply by filmo003** [reply]
Ah, that did it! Thanks! It may still be helpful to have a manifest.xml.sample file with the uuid removed in the future, but obviously no rush especially if you're traveling. Thanks again!

**Comment by osa1** [comment]
@filmo003 how did you get around the missing `Checker.fnt`?

**Reply by filmo003** [reply]
I just copied over the contents of one of the other .fnt files into it. I think it's a checker overlay used for the low power mode. Eventually we'll still need the actual Checker.fnt but this at least allows it to compile.
